exporter/collector/metricsexporter.go

```diff
@@ -162,6 +174,121 @@ func (m *metricMapper) metricToTimeSeries(
 	return timeSeries
 }
 
+func (m *metricMapper) exemplar(ex pdata.Exemplar) *distribution.Distribution_Exemplar {
+	attachments := []*anypb.Any{}
+	// TODO: Look into still sending exemplars with no span.
+	if !ex.TraceID().IsEmpty() && !ex.SpanID().IsEmpty() {
+		sctx, err := anypb.New(&monitoringpb.SpanContext{
+			// TODO - make sure project id is correct.
+			SpanName: fmt.Sprintf("projects/%s/traces/%s/spans/%s", m.cfg.ProjectID, ex.TraceID().HexString(), ex.SpanID().HexString()),
+		})
+		if err == nil {
+			attachments = append(attachments, sctx)
+		} else {
+			log.Printf("Failure to write span context: %v", err)
+		}
+	}
+	if ex.FilteredAttributes().Len() > 0 {
+		attr, err := anypb.New(&monitoringpb.DroppedLabels{
+			Label: attributesToLabels(ex.FilteredAttributes()),
+		})
+		if err == nil {
+			attachments = append(attachments, attr)
+		} else {
+			log.Printf("Failure to write filtered_attributes: %v", err)
+		}
+	}
+	return &distribution.Distribution_Exemplar{
+		Value:       ex.DoubleVal(),
+		Timestamp:   timestamppb.New(ex.Timestamp().AsTime()),
+		Attachments: attachments,
+	}
+}
+
+// Maps an exponential distribution into a GCM point.
+func (m *metricMapper) exponentialPoint(point pdata.ExponentialHistogramDataPoint) *monitoringpb.TypedValue {
+
+	growth := math.Exp2(math.Exp2(-float64(point.Scale())))
+	scale := math.Pow(growth, float64(point.Positive().Offset()))
+	// First calculate underflow bucket with all negatives + zeros.
+	underflow := point.ZeroCount()
+	for _, v := range point.Negative().BucketCounts() {
+		underflow += v
+	}
+
+	// Next, pull in remaining buckets.
+	counts := []int64{int64(underflow)}
```

Comment:
nit you can pre-allocate with `make([]int64, len[, cap])`. There should be `len(point.Positive().BucketCounts()) + 2` counts total

Reply:
Fixed